### PR TITLE
test: Enable Gradle tests with Gradle 7.5.1 and correct Spring Boot version (#14937) (CP: 23.3)

### PR DESCRIPTION
**Pull request by Artur-**
* test: extract slf4j version to a variable
* Use Gradle 7.5.1
* Fix Spring Boot test for newer Spring Boot

### CODE REVIEW

File: flow-plugins/flow-gradle-plugin/gradlew [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 #
-# Copyright � 2015-2021 the original authors.
+# Copyright © 2015-2021 the original authors.
```
Comment (vursen) [inline flow-plugins/flow-gradle-plugin/gradlew:4]: Please, remove the extra symbol.

File: flow-plugins/flow-gradle-plugin/gradlew [hunk 2]
```diff
@@ -32,10 +32,10 @@
 #       Busybox and similar reduced shells will NOT work, because this script
 #       requires all of these POSIX shell features:
 #         * functions;
-#         * expansions �$var�, �${var}�, �${var:-default}�, �${var+SET}�,
-#           �${var#prefix}�, �${var%suffix}�, and �$( cmd )�;
-#         * compound commands having a testable exit status, especially �case�;
-#         * various built-in commands including �command�, �set�, and �ulimit�.
+#         * expansions «$var», «${var}», «${var:-default}», «${var+SET}»,
```
Comment (vursen) [inline flow-plugins/flow-gradle-plugin/gradlew:35]: There is something wrong with these lines. A lot of weird `Â`.

File: flow-plugins/flow-gradle-plugin/gradlew (hunk 1 above)
Comment (Artur-) [reply flow-plugins/flow-gradle-plugin/gradlew:4]: It is exactly what ”gradle wrapper” generates

File: flow-plugins/flow-gradle-plugin/gradlew (hunk 2 above)
Comment (Artur-) [reply flow-plugins/flow-gradle-plugin/gradlew:35]: It is exactly what ”gradle wrapper” generates

File: flow-plugins/flow-gradle-plugin/gradlew (hunk 1 above)
Comment (vursen) [reply flow-plugins/flow-gradle-plugin/gradlew:4]: OK. Strange.

**Review by vursen — APPROVED** [review]

**Pull request merged**